test/parallel/test-dgram-multicast-set-interface-lo.js

```diff
@@ -0,0 +1,275 @@
+                        IFACE,
+                        MULTICAST,
+                        messagesNeeded.length,
+                        NOW]);
+      workers[worker.pid] = worker;
+
+      worker.messagesReceived = [];
+      worker.messagesNeeded = messagesNeeded;
+
+      //handle the death of workers
+      worker.on('exit', function(code, signal) {
+        // don't consider this the true death if the worker
+        // has finished successfully
+        // or if the exit code is 0
+        if (worker.isDone || code == 0) {
+          return;
+        }
+
+        dead += 1;
+        console.error('[PARENT] Worker %d died. %d dead of %d',
+                      worker.pid,
+                      dead,
+                      listeners);
+
+        if (dead === listeners) {
+          console.error('[PARENT] All workers have died.');
+          console.error('[PARENT] Fail');
+
+          killChildren(workers);
+
+          process.exit(1);
+        }
+      });
+
+      worker.on('message', function(msg) {
+        if (msg.listening) {
+          listening += 1;
+
+          if (listening === listeners) {
+            //all child process are listening, so start sending
+            sendSocket.sendNext();
+          }
+        } else if (msg.message) {
+          worker.messagesReceived.push(msg.message);
+
+          if (worker.messagesReceived.length === worker.messagesNeeded.length) {
+            done += 1;
+            worker.isDone = true;
+            console.error('[PARENT] %d received %d messages total.',
+                          worker.pid,
+                          worker.messagesReceived.length);
+          }
+
+          if (done === listeners) {
+            console.error('[PARENT] All workers have received the ' +
+                          'required number of ' +
+                          'messages. Will now compare.');
+
+            Object.keys(workers).forEach(function(pid) {
+              var worker = workers[pid];
+
+              var count = 0;
+
+              worker.messagesReceived.forEach(function(buf) {
+                for (var i = 0; i < worker.messagesNeeded.length; ++i) {
+                  if (buf.toString() === worker.messagesNeeded[i]) {
+                    count++;
+                    break;
+                  }
+                }
+              });
+
+              console.error('[PARENT] %d received %d matching messages.',
+                            worker.pid,
+                            count);
+
+              assert.equal(count, worker.messagesNeeded.length,
+                           'A worker received an invalid multicast message');
+            });
+
+            clearTimeout(timer);
+            console.error('[PARENT] Success');
+            killChildren(workers);
+          }
+        }
+      });
+    })(x);
+  }
+
+  var sendSocket = dgram.createSocket({
+    type: UDP[FAM],
+    reuseAddr: true
+  });
+
+  // don't bind the address explicitly for sending
```

Comment:
Extremely minor nit: can you upper case the first letter in the comments and add punctuation?